api/api.go

```diff
@@ -4,7 +4,7 @@ import (
 	"context"
 	"time"
 
-	"github.com/Azure/azure-sdk-for-go/profiles/latest/authorization/mgmt/authorization"
+	"github.com/Azure/azure-sdk-for-go/services/preview/authorization/mgmt/2018-01-01-preview/authorization"
```

Comment:
I see that there are [other versions under authorization/mgmt](https://github.com/Azure/azure-sdk-for-go/tree/main/services/preview/authorization/mgmt), is there a reason where going with this one in particular?

There is also a non-preview version for [2018-03-01](https://github.com/Azure/azure-sdk-for-go/tree/main/profiles/2018-03-01/authorization/mgmt/authorization) and [other newer ones](https://github.com/Azure/azure-sdk-for-go/tree/ec2ed6d8450ad53978efa96822bf9c9431a63ab4/profiles).

Reply:
From the [diff](https://github.com/hashicorp/vault-plugin-secrets-azure/pull/67/files) it appears we originally used `2018-01-01-preview` prior to merging https://github.com/hashicorp/vault-plugin-secrets-azure/pull/67. That PR switched to using `profiles/latest`, which appears to point to [2015-07-01](https://github.com/Azure/azure-sdk-for-go/blob/main/profiles/latest/authorization/mgmt/authorization/models.go#L15). I _think_ that explains why this error is showing up now.

Using `2018-01-01-preview` should be fine, but we might want to use something more recent as @calvn suggested. It looks like terraform-provider-azurerm is using [2020-04-01-preview](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/authorization/role_assignment_resource.go#L10).

Reply:
Good catch @austingebauer let's take a stepwise approach and upgrade back to what it was before, and we can look into bumping that up further afterwards (along with more testing). So upgrading to 2018-01-01-preview sounds good to me.